File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n=height.size();
        int start=0;
        int end=n-1;
        int maxarea=INT_MIN;
        while(start<=end)
        {
            if(height[start]<height[end])
            {
                maxarea=max(maxarea,(end-start)*height[start]);
                start++;
            }
            else
            {
                maxarea=max(maxarea,(end-start)*height[end]);
                end--;
            }
            
        }
        return maxarea;
    }
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i=0
        j=len(height)-1
        res=0
        while i<j:
            a=min(height[i],height[j])
            w=j-i
            if a*w > res:
                res=a*w
            if height[i]<height[j]:
                i+=1
            else:
                j-=1
           
        return res

        